internal/host/plugin/repository_host_catalog.go

```diff
@@ -454,6 +458,36 @@ func (r *Repository) UpdateCatalog(ctx context.Context, c *HostCatalog, version
 				recordUpdated = true
 			}
 
+			if needSetSync {
+				// We also need to mark all host sets in this catalog to be
+				// synced as well.
+				setsForCatalog, _, err := r.getSets(ctx, "", returnedCatalog.PublicId)
+				if err != nil {
+					return errors.Wrap(ctx, err, op, errors.WithMsg("unable to get sets for host catalog"))
+				}
+
+				for _, set := range setsForCatalog {
+					newSet := set.clone()
+					newSet.NeedSync = true
+					currentVersion := newSet.Version
+					newSet.Version = currentVersion + 1
+					var msg oplog.Message
+					n, err := w.Update(ctx, newSet, []string{"NeedSync", "version"}, []string{}, db.NewOplogMsg(&msg), db.WithVersion(&currentVersion))
+					if err != nil {
+						return errors.Wrap(ctx, err, op, errors.WithMsg("unable to update host set"))
+					}
+
+					if n != 1 {
```

Comment:
Some of the sets might already have NeedSync = true which would cause it to not be updated I don't think.  It might be better for this conditional to be updated.
```suggestion
					if n > 1 {
```